test/e2e/e2e_test.go

```diff
@@ -366,8 +366,8 @@ func newFixture(ns string, t *testing.T) *fixture {
 }
 
 func (f *fixture) targetStep(step int, relName string) {
-	patch := fmt.Sprintf(`{"spec": {"targetStep": %d}}`, step)
-	_, err := shipperClient.ShipperV1().Releases(f.namespace).Patch(relName, types.MergePatchType, []byte(patch))
+	patch := fmt.Sprintf(`[{ "op": "replace", "path": "/spec/targetStep", "value": %d }]`, step)
```

Comment:
I know we talked about this, but I forgot -- my understanding was that only **Strategic**Merge was unsupported for CRDs, but Merge patches are OK. The reason I think this is because this is how I patch things with kubectl:

```
$ kubectl patch rel web-serving-excellence-f4ef7212-0 --type=merge -p '{"spec": {"targetStep": 1}}' 
release.shipper.booking.com/web-serving-excellence-f4ef7212-0 patched
```

And when I run that with a high log level, this is the API call that `kubectl` is making:
```
I0912 00:10:34.003942   34637 request.go:897] Request Body: {"spec": {"targetStep": 1}}
I0912 00:10:34.004028   34637 round_trippers.go:386] curl -k -v -XPATCH  -H "Accept: application/json" -H "Content-Type: application/merge-patch+json" -H "User-Agent: kubectl/v1.11.1 (darwin/amd64) kubernetes/b1b2997" 'https://192.168.99.100:8443/apis/shipper.booking.com/v1/namespaces/default/releases/web-serving-excellence-f4ef7212-0'
```

Reply:
This has been reverted.